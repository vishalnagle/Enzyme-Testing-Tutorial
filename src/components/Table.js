import React,{useState} from 'react'

const Table = () => {

  return (
    <table className='custom-border'>
        <thead>
            <tr>
                <th>Names</th>
                <th>Email</th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td>Demo</td>
                <td>vishal@test.com</td>
            </tr>
        </tbody>
    </table>
  )
}

export default Table