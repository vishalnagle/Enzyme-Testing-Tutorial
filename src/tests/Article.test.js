import Article from "../components/Article";
import Table from "../components/Table";
import { shallow,mount } from "enzyme";


describe("Article page",()=>{

    test("click count",()=>{
        let wrapper = shallow(<Article/>)
        expect(wrapper.find(".clicks-0").length).toEqual(1)
    })

    test("click button",()=>{
        let wrapper = shallow(<Article/>)
        wrapper.find("button").simulate("click")
    })

    test("using dive",()=>{
        let wrapper = shallow(<Article/>)
        expect(wrapper.find(Table).dive().find(".custom-border").length).toBe(1)
    })

    test("using snapshot",()=>{
        let wrapper = shallow(<Table/>)
        expect(wrapper).toMatchSnapshot()
    })

})

describe("Mount vs shallow",()=>{
    test("logging shallow",()=>{
        let wrapper = shallow(<Article/>)
        console.log("shallow",wrapper.debug())
    })

    test("logging mount",()=>{
        let wrapper = mount(<Article/>)
        console.log("mount",wrapper.debug())
    })
})