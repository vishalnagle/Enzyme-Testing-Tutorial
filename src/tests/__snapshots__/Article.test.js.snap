// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article page using snapshot 1`] = `
<table
  className="custom-border"
>
  <thead>
    <tr>
      <th>
        Names
      </th>
      <th>
        Email
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Demo
      </td>
      <td>
        vishal@test.com
      </td>
    </tr>
  </tbody>
</table>
`;
